TYPO3.Flow/Classes/TYPO3/Flow/Http/Component/TrustedProxiesComponent.php

```diff
@@ -50,14 +50,21 @@ public function handle(ComponentContext $componentContext)
         }
 
         $hostHeader = $this->getFirstTrustedProxyHeaderValue(self::HEADER_HOST, $trustedRequest);
+        $portFromHost = null;
         if ($hostHeader !== null) {
+            $portSeparator = strrpos($hostHeader, ':');
+            if ($portSeparator > 0) {
```

Comment:
Ok, actually the host header might include an IP address instead of the domain name and now think what happens when that is an IPv6 address... lots of `:` involved...